### ISSUE TITLE
execute PromQL:eseal_num_total @ time()-(time()+28800)%86400，the query syntax is wrong

### ISSUE THREAD

**Issue by JaminVan** [prometheus/prometheus#11918]
### What did you do?

When I execute PromQL:eseal_num_total @ time()-(time()+28800)%86400, it prompts that the query syntax is wrong, how can I query the metric at 0:00 of the current day，can anyone help me, thank you very much

### What did you expect to see?

eseal_num_total{instance="10.1.1.101", job="test_2", ownerType="dept"} | 4
eseal_num_total{instance="10.1.1.101", job="test_2", ownerType="user"} | 4

### What did you see instead? Under which circumstances?

When I execute PromQL in the prometheus query console: eseal_num_total @ time()-(time()+28800)%86400, the web page prompts: Error executing query: invalid parameter "query": 1:27: parse error: unexpected identifier "time " in @, expected timestamp,but when I replaced it with: eseal_num_total @ timestamp(round(vector(0+time()-(time()+28800)%86400))), and executed, I got this error message: Error executing query: invalid parameter "query": 1:27: parse error: unexpected identifier "timestamp" in @, expected timestamp

### System information

Linux  4.19.0-10-amd64 #1 SMP Debian 4.19.132-1  x86_64

### Prometheus version

```text
Version	2.32.1
Revision	41f1a8125e664985dd30674e5bdf6b683eff5d32
Branch	HEAD
BuildUser	root@54b6dbd48b97
BuildDate	20211217-22:08:06
GoVersion	go1.17.5
```

**Comment by damnever** [comment]
I believe this feature will be useful.

**Comment by schettgen** [comment]
I would need the same 

I have the usecase we use a third party service to get data. 
We have a subscription which allows us to sent e.g. 5000 requests per Month 
Counted from fist of month till last day of month.

Now I have a counter which counts the sent requests.

I try now to create a alert in case we reach 75% of our budget 

to do this I would need the value of the counter at the first day of the month 

I tried following
```
timestamp(floor(vector(time()) - (day_of_month() * 86400000) )) 
```
which returns `1715066219.26`


so I tried 
```
xxxxxxxxxxxxxxxxxx_api_counter{category="httpRequest", event="requestToApi"} @ timestamp(floor(vector(time()) - (day_of_month() * 86400000) )) 
```
which returns error 
`Error executing query: 1:80: parse error: unexpected identifier "timestamp" in @, expected timestamp`
